### PR TITLE
First go at findbugs cleanup. Got all the simple ones.

### PR DESCRIPTION
**Pull request by ahaarrestad**
A first run to start working on #200. Removed some dead code (containing bugs etc) and fixing the bugs I could fix with confidence. 
After this patch 5 bugs remains. 

1 scary
3 troubling
1 of concern

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@ahaarrestad__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by ahaarrestad** [comment]
removed a bit of dead code as I was going through the code. A bit of cleanup is allways good :-)

File: src/main/java/microsoft/exchange/webservices/data/AutodiscoverEndpoints.java [hunk 1]
```diff
@@ -15,47 +15,11 @@
...
-  /**
-   * No endpoints available.
-   */
-  None(0),
+  None,
+  Legacy,
+  Soap,
+  WsSecurity,
+  WSSecuritySymmetricKey,
+  WSSecurityX509Cert;
```
Comment (serious6) [inline src/main/java/microsoft/exchange/webservices/data/AutodiscoverEndpoints.java]: why did you remove the javadoc here? They where not very declarative but maybe just the content would need to be changed.

File: src/main/java/microsoft/exchange/webservices/data/AttachmentCollection.java [hunk 2]
```diff
@@ -379,7 +377,9 @@ public void validate() throws Exception {
             contactPhotoFound = true;
           }
         }
-        attachment.validate(attachmentIndex);
+        if (null != attachment) {
```
Comment (serious6) [inline src/main/java/microsoft/exchange/webservices/data/AttachmentCollection.java]: this check is already mentioned in [L357](https://github.com/ahaarrestad/ews-java-api/blob/FirstFindbugsCleanup/src/main/java/microsoft/exchange/webservices/data/AttachmentCollection.java#L357)

**Comment by serious6** [comment]
please add findbugs to the report-module in pom.xml:
```xml
    <reporting>
        <plugins>
           [...]
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>findbugs-maven-plugin</artifactId>
                <version>3.0.0</version>
            </plugin>
        </plugins>
    </reporting>
```
this report then can be seen if you run `mvn clean site`

**Comment by serious6** [comment]
Some suggestions:

* please add findbugs to the report-module in pom.xml (this will generate a report when `mvn site` is called):
```xml
    <reporting>
        <plugins>
           [...]
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>findbugs-maven-plugin</artifactId>
                <version>3.0.0</version>
            </plugin>
        </plugins>
    </reporting>
``` 
* I think we should also add the findbugs-plugin as a suggested plugin. In the `CONTRIBUTING.md`.
* There should be a check added to the travis.yml
```yaml
after_success:
  - mvn findbugs:check
```

File: src/main/java/microsoft/exchange/webservices/data/AutodiscoverEndpoints.java (hunk 1 above)
Comment (ahaarrestad) [inline src/main/java/microsoft/exchange/webservices/data/AutodiscoverEndpoints.java]: In my oppinion, comments that add no extra information, only boats the files with extra lines, are noise. I left the comments in one enum, because they included more information than the enum, but a comment /** the monday **/ for an enum "monday" just clutters the file.

**Comment by evpaassen** [comment]
I'm trying to review the changes, but that's quite hard because some files (or portions of files) have been reformatted with incorrect (seemingly unlimited) line length. Could you try to fix that?

**Comment by ahaarrestad** [comment]
@serious6: I've added findbugs to pom and travis.yml. Also a suggested text for contributing.md

@evpaassen: I think I've found all the lines that were autoformatted to be too long. Please let me know if I missed any.

File: src/main/java/microsoft/exchange/webservices/data/AttachmentCollection.java (hunk 2 above)
Comment (ahaarrestad) [inline src/main/java/microsoft/exchange/webservices/data/AttachmentCollection.java]: I think that will be even more confusing...

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -10,6 +10,8 @@ There are several ways to contribute to the project:
 
 GitHub supports [markdown](http://github.github.com/github-flavored-markdown/), so when filing bugs make sure you check the formatting before clicking submit.
 
+ews-java-api uses findbugs to check the code, and you should do so too before submitting your code. A findbugs report can be generated using "mvn site", or by downloading it as a plugin for your favorite IDE.
```
Comment (serious6) [inline CONTRIBUTING.md]: maybe this should be added to an independent section as there might join more suggested plugins

**Comment by serious6** [comment]
travis build fails on:
- findbugs
  -  [openjdk6](https://travis-ci.org/OfficeDev/ews-java-api/jobs/49887947#L1702) maybe these tests cant be run under openjdk1.6
- cobertura
  -  [all builds](https://travis-ci.org/OfficeDev/ews-java-api/jobs/49887947#L1751)

**Comment by ahaarrestad** [comment]
Cobertura is probably not registeres for ews-java-api: https://coveralls.io/sign-up ?
where do you see the failed builds?

**Comment by serious6** [comment]
You  have to expand the corbatura-section in the build-log

**Comment by ahaarrestad** [comment]
thank's. found it. trying to figure out how to exclude findbugs for java6......

**Comment by ahaarrestad** [comment]
Updated and now I think both cobertura and findbugs is running. but I need somebody with login for ews-java-api to look at https://coveralls.zendesk.com/hc/en-us#random-commits to enable coveralls.

**Comment by serious6** [comment]
Maybe we should skip coveralls until we have a more proper coverage?  Don't think anybody wants this on the dashboard now.

File: pom.xml [hunk 4]
```diff
@@ -66,10 +67,20 @@
             <!-- handle broken builds on jdk1.8 due to doclint function -->
             <id>default-jdk18-profile</id>
             <activation>
-                <jdk>[1.8,)</jdk>
+                <jdk>1.8</jdk>
```
Comment (serious6) [inline pom.xml]: I tried this notation but it was not working with my latest maven installed. Have you checked the profile to be working?

Comment (ahaarrestad) [inline pom.xml]: I have reverted this change after figuring out what went wrong :-)

**Comment by ahaarrestad** [comment]
removed coveralls until we have better test-coverage

File: src/main/java/microsoft/exchange/webservices/data/TimeZoneDefinition.java [hunk 5]
```diff
@@ -149,7 +136,7 @@ protected void readAttributesFromXml(EwsServiceXmlReader reader)
     if (this.id == null || this.id.isEmpty()) {
       String nameValue = (this.getName() == null || this.
           getName().isEmpty()) ? "" : this.getName();
-      this.setId(NoIdPrefix + Math.abs(nameValue.hashCode()));
+      this.setId(NoIdPrefix + Integer.toString(nameValue.hashCode()).replace("-", ""));
```
Comment (serious6) [inline src/main/java/microsoft/exchange/webservices/data/TimeZoneDefinition.java]: What is this about?

Comment (ahaarrestad) [inline src/main/java/microsoft/exchange/webservices/data/TimeZoneDefinition.java]: Math.abs(Integer.MAX) does not work. this workaround does not change the workings, but is quite ugly :-)

Comment (ahaarrestad) [inline src/main/java/microsoft/exchange/webservices/data/TimeZoneDefinition.java]: Updated to BigInteger.abs()

**Comment by serious6** [comment]
I think the ID-generation should not be part of this PR and needs to have unittests included as well. Maybe we can have this splitted.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -10,6 +10,9 @@ There are several ways to contribute to the project:
 
 GitHub supports [markdown](http://github.github.com/github-flavored-markdown/), so when filing bugs make sure you check the formatting before clicking submit.
 
+### plugins used
+* findbugs: ews-java-api uses findbugs to check the code, and you should do so too before submitting your code. A findbugs report can be generated using "mvn site", or by downloading it as a plugin for your favorite IDE.
```
Comment (serious6) [inline CONTRIBUTING.md]: please add a link to the findbugs plugin page.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -10,6 +10,9 @@ There are several ways to contribute to the project:
 
 GitHub supports [markdown](http://github.github.com/github-flavored-markdown/), so when filing bugs make sure you check the formatting before clicking submit.
 
+### plugins used
```
Comment (serious6) [inline CONTRIBUTING.md]: **Plugins used**

**Comment by serious6** [comment]
please split this PR:
* adding of the maven-plugins and documentation stuff
* fixing the checkstyle issues
   * maybe you should provide more information on what has been fixed. 
   * commits should be made for every fixed issue

**Comment by serious6** [comment]
@ahaarrestad are you planning any changes?

**Comment by ahaarrestad** [comment]
Yes. I'll follow up as soon as I manage :-)

**Comment by vbauer** [comment]
:+1:

**Comment by serious6** [comment]
@ahaarrestad please reopen when you had the time for refactoring.
